src/tokenizers/config.rs

```diff
@@ -0,0 +1,47 @@
+use crate::error::BoxedError;
+use std::path::PathBuf;
+
+use snafu::Snafu;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(super)))]
+pub enum TokenizerConfigError {
+    #[snafu(display("Couldn't resolve filepaths in tokenizer configuration"))]
+    ResolvePaths { source: BoxedError },
+
+    #[snafu(display("Cannot open tokenizer config for reading: {path:?}"))]
+    OpenConfigFile {
+        path: PathBuf,
+        source: std::io::Error,
+    },
+
+    #[snafu(display("Cannot deserialize tokenizer JSON configuation"))]
+    DeserializeConfigJSON { source: serde_json::Error },
+}
+
+/// Trait for tokenizer configs that have paths to local files.
+pub trait TokenizerConfigWithPaths {
+    /// Resolves all file paths in the tokenizer config.
+    ///
+    /// * `resolver` - A function that takes a filename and returns a resolved path.
+    fn resolve_paths<'a, F>(&'a mut self, resolver: F) -> Result<(), TokenizerConfigError>
```

Comment:
Not sure if I'm a fan of the closure passing style. Seems like this needs something similar to the repository abstraction of Curated Transformers?

I am also don't understand why the `Default` plus `resolve_paths` combo is needed? Couldn't we have a generic constructor for a tokenizer? Like:

```rust
trait FromRepo {
    fn from_repo(repo: &Repo) -> Self;
}
```

or something along those lines? Then the HF Hub implementation could call `from_repo` with HF Hub as a repository. Or rather, there would be a `FromRepo` trait that does the tokenizer construction from any repo and then `FromHFHub` would be a simple wrapper that calls `from_repo` with a HF Hub repo.

Reply:
Indeed, I'm not a fan either. The repository abstraction from CT would definitely makes more sense and fits here better. I can either do that in a separate PR or modify this one, but the former might make more sense as the abstraction should also apply to models.

Reply:
I would just implement it for tokenizers first, I can hold off doing a PR for Llama and NeoX until the repo abstraction is there. I still have a bunch of other things that I need to work on/fix (like support for passing positions) and causal LMs.